src/pages/index.js

```diff
@@ -7,36 +7,95 @@ import UserInfo from "../components/UserInfo.js";
 import { initialCards } from "../utils/constants.js";
 import { config } from "../utils/constants.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
+import Popup from "../components/Popup.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+
+let section;
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "953ea6cd-2c01-4d7b-be00-e077aa224921",
+    "Content-Type": "application/json",
+  },
+});
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    setIsLiked,
+    handleDelete
+  );
   const cardElement = card.getCardView();
+
   return cardElement;
 }
 
+function setIsLiked(card) {
+  const cardId = card._cardData._id;
+
+  if (card.isLiked()) {
+    api.dislikeCard(cardId).then((res) => {
+      card.setIsLiked(false);
+    });
+  } else {
+    api.likeCard(cardId).then((res) => card.setIsLiked(true));
+  }
+}
+
 function handleImageClick(cardData) {
   popupWithImage.open(cardData);
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-
-section.renderItems();
+function handleDelete(card) {
+  popupConfirmDeleteCard.open();
+  const confirmButton = document.querySelector("#confirm-delete-button");
+  confirmButton.addEventListener("click", () => {
+    api
```

Comment:
Needs correction: don't forget to notify the user about loading by changing the confirm button text